module3/lessons/Week2/RestfulRoutes.md

```diff
@@ -0,0 +1,174 @@
+---
+layout: page
+title:  RESTful Routes
+---
+
+## Learning Objectives
+* Review CRUD functionality
+* Understand REST as it relates to CRUD
+* Learn the 7 RESTful routes
+* Understand the different between RESTful and non-RESTful routes
+
+## But first...
+
+<details>
+    <summary>
+    What is CRUD again?
+    </summary>
+
+    Thus far, we've used CRUD functionality to **C**reate, **R**ead, **U**pdate, and **D**elete from a database.
+</details>
+
+<details>
+    <summary>
+    What is an HTTP route?
+    </summary>
+
+    An HTTP route is the code responsible for receiving and responding to an HTTP request.
+</details>
+
+<details>
+    <summary>
+    What is a RESTful route?
+    </summary>
+
+    <ul>
+        <li>A RESTful route follows a common pattern to map between HTTP routes/methods and CRUD functionality.</li>
+        <li>By the way, REST stands for **RE**presentation **S**tate **T**ransfer. This may be an interview question; remember it, just in case.</li>
+    </ul>
+</details>
+
+<details>
+    <summary>
+    What is a URI?
+    </summary>
+
+    URI stands for **U**niform **R**esource **I**dentifier. It is the part of the URL after the domain. For our purpose, the URI is the same as our route.
+    <br><br>
+    For example, for the URL <code>https://en.wikipedia.org/wiki/Uniform_Resource_Identifier</code>:
+    <ul>
+        <li> the domain is <code>en.wikipedia.org</code> </li>
+        <li> the URI is <code>/wiki/Uniform_Resource_Identifier</code></li>
+    </ul>
+</details>
+
+## The List
+
+Here is a table of the list of RESTful routes. The given URI is based on the context from our `MvcMovie` application.
+
+| Route Name | URI | HTTP Method | CRUD | Return Type | Purpose |
```

Comment:
I really like having all the routes displayed in a table! Makes it easy to see all of the options.

Reply:
I agree!